src/base_model.py

```diff
@@ -70,3 +58,20 @@ def to_dict(self) -> dict:
         :return: the dictionary
         """
         return self.Schema().dump(self)
+
+    class Schema(marsh.Schema):
+        """The marshmallow schema all model schemas have to inherit from."""
+
+        id = marsh.fields.UUID(dump_only=True)
+        created_at = marsh.fields.DateTime(format="iso", dump_only=True)
+        updated_at = marsh.fields.DateTime(format="iso", dump_only=True)
+
```

Comment:
```suggestion
@abc.abstractmethod
def _dump(self, obj: 'SerializableBaseModel') -> dict:
    raise NotImplementedError()

@marsh.pre_dump         
def dump(self, obj: 'SerializableBaseModel', **_) -> dict:
    return self._dump(obj)
```

As you mentioned via Slack you want the funktionality of `make` the other way around. You can try my suggestion. Please validate if it really works.

You can read more about it [here](https://marshmallow.readthedocs.io/en/stable/extending.html) and [here](https://marshmallow.readthedocs.io/en/stable/marshmallow.decorators.html#marshmallow.decorators.pre_dump).

Reply:
You would then override `_dump` in the subclasses to specify the serialization behavior.

Reply:
I'd stay with overwriting `to_dict()` function if needed. We'll have some classes where `Schema().dump()` is completely appropriate and sometimes not. That's why I won't add an abstract method `dump()` with a decorator and the abstract method `_dump()`. 🙃

Reply:
If I understand it correctly, `_dump()` would take the object as input and return the dictionary?

Reply:
It's ok I didn't like the tests anyway :D

My inner linter doesn't like that serialization is achieved with a different method then deserialization. But considering our time constraints that is totally ok :)

Reply:
> My inner linter doesn't like that serialization is achieved with a different method then deserialization.

same here 🤔

Reply:
I'm ok with ignoring it

Reply:
I'm unsure what happens when the dumping method doesn't find `platform_ids` list in the class serialize.

Reply:
There are platform ids. A `ScheduleConfigruation`-object has a  `train_schedule_platform_references` field. It gets that field by the Foreign Key definition in `ScheduleConfigurationXSimulationPlatform`.

Reply:
Yes I understand, but it's not called `platform_ids` :/ I'm worried that the Schema is confused, if it doesn't find anything at `platform_ids`. arggghhh
And  train_schedule_platform_references doesn't hold the ids. Instead it holds the object containing the id ;)

Reply:
Yes you are right. I'm sure there is a way to make it elegant and "right". But that would definitely take more time. In the end it is your decision. I am fine with the faster alternative.

Reply:
Alright. I'll try something later, but for now i'll leave it this way 🙃 
Thank you for the discussion, your research and comments! 💚